src/components/input/input.js

```diff
@@ -679,16 +694,17 @@ function mdMaxlengthDirective($animate, $mdUtil) {
       });
     });
 
-    function renderCharCount(value) {
-      // If we have not been initialized or appended to the body yet; do not render
+    function renderCharCount() {
+      // If we have not been initialized or appended to the body yet; do not render.
       if (!charCountEl || !charCountEl.parent) {
-        return value;
+        return;
       }
 
+      var elementVal = element.val();
+      elementVal = ngTrim && angular.isString(elementVal) ? elementVal.trim() : elementVal;
```

Comment:
Nit: AngularJS ignores `ngTrim` for inputs of type `password` (and ever trim the value). In such a case, there will be an inconsistency between the value reported by `mdMaxlength` and the `$modelValue` 😛

Reply:
Ah yeah, I saw that but didn't imagine the case where a password input would be used with `mdMaxlength`. I think it makes sense to support that. Thanks!